File: 国王游戏(简化版).cpp
#include <bits/stdc++.h>
using namespace std;

#define ff first
#define ss second
#define endl '\n'
#define int long long
#define PII pair<int, int>

void solve() {
    int n, ans = 0, now = 1;
    cin >> n;
    vector<PII> ab(n + 1);
    for (int i = 0; i <= n; ++i) {
        cin >> ab[i].ff >> ab[i].ss;
    }
    sort(ab.begin(), ab.end(), [](const PII &a, const PII &b) {
        return a.ff * a.ss < b.ff * b.ss;
    });
    for (int i = 0; i <= n; ++i) {
        ans = max(ans, now / ab[i].ss);
        now *= ab[i].ff;
    }
    cout << ans << endl;
}

signed main() {
    cin.tie(nullptr)->sync_with_stdio(false);
    cout << fixed << setprecision(15);
    solve();
    return 0;
}

File: 国王游戏(简化版).py
import sys
input = lambda: sys.stdin.readline().strip()
mod = 10 ** 9 + 7 # 998244353

n = int(input())
ab = [tuple(map(int, input().split())) for _ in range(n + 1)]
ans, now = 0, 1
ab.sort(key = lambda x : x[0] * x[1])
for a, b in ab:
    ans = max(ans, now // b)
    now *= a
print(ans)
